scripts/zones/Sacrarium/mobs/Mariselles_Pupil.lua

```diff
@@ -0,0 +1,188 @@
+        {100.2014, -2.0000, -115.8586},
+        {112.5715, -2.0000, -115.8909},
+        {111.0784, -2.0000, -109.7074},
+        {97.4817, -2.0000, -109.9677},
+        {88.0128, -2.0000, -110.1289}
+    }
+    NW = -- Sorthwest room
+    {
+        {9.7012, -2, 109.5587},
+        {22.042, -2, 110.0485},
+        {31.9411, -2, 112.7432},
+        {30.6844, -2, 117.8574},
+        {18.2341, -2, 116.4862},
+        {7.4438, -2, 116.2394},
+        {9.3145, -2.1785, 130.6581},
+        {19.8528, -2.7298, 130.7574},
+        {30.7859, -2.1785, 129.5448}
+    }
+    NM = --Sorthmid room
+    {
+        {70.8789, -2, 109.542},
+        {60.4267, -2, 110.0887},
+        {48.2896, -2, 111.3439},
+        {49.2171, -2, 117.5056},
+        {61.5946, -2, 116.0528},
+        {71.7462, -2, 116.0356},
+        {69.0619, -2.1785, 130.0243},
+        {58.1095, -2.7298, 129.9273},
+        {50.1262, -2.1785, 130.2446}
+    }
+    NE = --Sortheast room
+    {
+        {88.6123, -2, 110.1267},
+        {99.8369, -2, 109.8693},
+        {111.5963, -2, 111.03},
+        {112.236, -2, 117.1377},
+        {88.3052, -2, 118.776},
+        {88.8908, -2.1785, 129.5286},
+        {100.0305, -2.7298, 130.7616},
+        {110.8072, -2.1785, 130.0085},
+        {100.0483, -2, 116.4553}
+    }
+
+    local teleTime = mob:getLocalVar("teleTime")
+    if mob:getBattleTime() - teleTime > 30 and mob:getBattleTime() > 59 then
+        local profLocation = mob:getLocalVar("spawnLocation")
+        print(profLocation)
+        if profLocation == 0 then
+            randPos1 = NE[math.random((1), (9))]
+            Teleport(mob, 2000, randPos1)
+            mob:setSpawn(randPos1[1], randPos1[2], randPos1[3])
+        elseif profLocation == 1 then
+            randPos1 = NM[math.random((1), (9))]
+            Teleport(mob, 2000, randPos1)
+            mob:setSpawn(randPos1[1], randPos1[2], randPos1[3])
+        elseif profLocation == 2 then
+            randPos1 = NW[math.random((1), (9))]
+            Teleport(mob, 2000, randPos1)
+            mob:setSpawn(randPos1[1], randPos1[2], randPos1[3])
+        elseif profLocation == 3 then
+            randPos1 = SE[math.random((1), (9))]
+            Teleport(mob, 2000, randPos1)
+            mob:setSpawn(randPos1[1], randPos1[2], randPos1[3])
+        elseif profLocation == 4 then
+            randPos1 = SM[math.random((1), (9))]
+            Teleport(mob, 2000, randPos1)
+            mob:setSpawn(randPos1[1], randPos1[2], randPos1[3])
+        else
+            randPos1 = SW[math.random((1), (9))]
+            Teleport(mob, 2000, randPos1)
+            mob:setSpawn(randPos1[1], randPos1[2], randPos1[3])
+        end
+        mob:setLocalVar("teleTime", mob:getBattleTime())
+    end
+end
+
+entity.onMobDisengage = function(mob)
+    mob:setLocalVar("teleTime", 0)
+end
+
+entity.onMobRoam = function(mob)
+    local profLocation = mob:getLocalVar("spawnLocation")
+    print(profLocation)
+    if profLocation == 0 then
```

Comment:
Same as above comment. With a table of numeric values as key, this could be simplified

Reply:
Check now